File: sesion5-lab/spring-security-audit/audit-server/src/main/java/com/example/audit/service/VulnerableAuditService.java
package com.example.audit.service;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.stereotype.Service;

@Service
public class VulnerableAuditService {

    private static final Logger log = LoggerFactory.getLogger(VulnerableAuditService.class);

    public boolean login(String username, String password) {
        boolean ok = "ana".equals(username) && "Secr3t!".equals(password);
        // ANTES: texto libre, sin correlation ID, incluye password
        log.info("User {} tried login with password {} result={}", username, password, ok ? "OK" : "FAIL");
        return ok;
    }

    public void logAccess(String username, String path, int status) {
        log.info("Access user={} path={} status={}", username, path, status);
    }
}

File: sesion5-lab/aspnet-security-audit/audit-demo/Services/VulnerableAuditService.cs
namespace AuditDemo.Services;

using AuditDemo.Support;

public class VulnerableAuditService(LogCapture logCapture)
{
    public bool Login(string username, string password)
    {
        var ok = username == "ana" && password == "Secr3t!";
        var line = $"User {username} tried login with password {password} result={(ok ? "OK" : "FAIL")}";
        logCapture.Add(line);
        return ok;
    }

    public void LogAccess(string username, string path, int status)
    {
        logCapture.Add($"Access user={username} path={path} status={status}");
    }
}
